coordinator/src/main/scala/filodb.coordinator/queryengine2/QueryEngine.scala

```diff
@@ -198,7 +198,17 @@ class QueryEngine(dataset: Dataset,
                                       submitTime: Long,
                                       options: QueryOptions,
                                       lp: LabelValues): Seq[LabelValuesExec] = {
-    shardsFromFilters(lp.filters, options).map { shard =>
+    var shards: Seq[Int] = Seq.empty
+    try {
+      shards = shardsFromFilters(lp.filters, options)
+    } catch {
+      // expected - filters don't have enough info to derive shard mapping, send query to all the shards
+      // reusing the existing logic
+      case e: BadQueryException =>
+        logger.error("exception while materializing the plan", e)
```

Comment:
If this is expected, is this an error or warn? Or info?

Reply:
@broneill ack, updated to warn